packages/firestore/src/api/user_data_converter.ts

```diff
@@ -574,7 +527,7 @@ export class UserDataConverter {
         // No transform to add for a delete, so we do nothing.
       } else if (context.dataSource === UserDataSource.Update) {
         assert(
-          context.path == null || context.path.length > 0,
+          context.path.length > 0,
```

Comment:
Note that this is safe since at the top of the method we do:

```
    if (context.path === null) {
      throw context.createError(
        `${value.methodName} is not currently supported inside arrays`
      );
    }
```